### ISSUE TITLE
Bug: particular input always causes error

### ISSUE THREAD

**Issue by KeisukeNagakawa** [codders/dyad-finder#35]
Hi, first of all, thank you very much for this great library.

When I calculate  inputs as below:
```typescript
 solver.solve({
      student_prefs: [
        {
          A: ["D", "E"],
          B: ["E", "F"],
          C: ["D", "A"],
          D: ["E", "B"],
          E: ["F", "A"],
          F: ["C", "E"],
        },
      ],
    })
```

it always causes error saying:
```
Error: TypeError: Cannot read properties of undefined (reading 'preferences')
    at calculate_next_loop_entry (/Users/kay/Code/aeta-event/functions/src/utils/srmSolver.ts:122:11)
    at find_loop (/Users/kay/Code/aeta-event/functions/src/utils/srmSolver.ts:159:18)
    at solve_for_stable (/Users/kay/Code/aeta-event/functions/src/utils/srmSolver.ts:194:18)
    at Module.solve (/Users/kay/Code/aeta-event/functions/src/utils/srmSolver.ts:259:20)
    at /Users/kay/Code/aeta-event/functions/src/__tests__/fixed.test.ts:17:29
    at /Users/kay/Code/aeta-event/functions/node_modules/vitest/dist/chunk-runtime-chain.37ec5d73.js:2297:13
    at /Users/kay/Code/aeta-event/functions/node_modules/vitest/dist/chunk-runtime-chain.37ec5d73.js:2172:26
    at runTest (/Users/kay/Code/aeta-event/functions/node_modules/vitest/dist/chunk-runtime-setup.ab6b6274.js:417:42)
    at async runSuite (/Users/kay/Code/aeta-event/functions/node_modules/vitest/dist/chunk-runtime-setup.ab6b6274.js:512:15)
    at async runSuite (/Users/kay/Code/aeta-event/functions/node_modules/vitest/dist/chunk-runtime-setup.ab6b6274.js:512:15)
    at async runFiles (/Users/kay/Code/aeta-event/functions/node_modules/vitest/dist/chunk-runtime-setup.ab6b6274.js:646:5)
    at async startTestsNode (/Users/kay/Code/aeta-event/functions/node_modules/vitest/dist/chunk-runtime-setup.ab6b6274.js:664:3)
    at async /Users/kay/Code/aeta-event/functions/node_modules/vitest/dist/entry.js:96:11
    at async Module.withEnv (/Users/kay/Code/aeta-event/functions/node_modules/vitest/dist/chunk-runtime-setup.ab6b6274.js:162:5)
    at async run (/Users/kay/Code/aeta-event/functions/node_modules/vitest/dist/entry.js:89:7)
    at async file:///Users/kay/Code/aeta-event/functions/node_modules/tinypool/dist/esm/worker.js:109:20
```

Any idea why it fails? 

 I suppose `qidash` is `undefined` because its preferences has an array of length 1  but `.preferences[1]` is called...

Thanks!

**Comment by KeisukeNagakawa** [comment]
maybe  [this line](https://github.com/codders/dyad-finder/blob/master/functions/src/solver.ts#L123) of code should be changed as follows?

```typescript
  const qidash = state[current_entry[0]].preferences[0];
```

I'm not checking its logical validity but this change blowed out errors above.

**Comment by codders** [comment]
Dear Nagakawa-san,

Thank you for your input, and sorry that you encounter this bug. It's a while since I looked at the project :)

I will take a look in the coming days (and will try and update the outdated libraries), but your fix sounds like a good start.

Can I ask where you found the project?

Arthur

**Comment by KeisukeNagakawa** [comment]
Dear Sie @codders ,

Thank you too for your surprisingly fast response! I'm not in a hurry so please take some time to check it.

>Can I ask where you found the project?

I found this project from [a link in wikipedia's citation](https://en.wikipedia.org/wiki/Stable_roommates_problem#cite_ref-8))!

**Comment by codders** [comment]
Dear Nagakawa-san,

Thank you again for your input and test-case. I have found there was an error in the implementation of the algorithm - I removed the error and the algorithm should now return a result for this input.

The fix you proposed would have changed the logic of the algorithm. For a 'stable' set of preferences, there will always be at least 2 preferences per user (so preferences.length will always be at least 2 and preferences[1] will always exist). Unfortunately, in my implementation of the loop reduction, I had only removed preferences in one direction (and not symmetrically), which it seems was causing this problem.

I have also updated the code now to a newer version of node, and updated the website deployment so that the code now runs again normally.

Many thanks!

Arthur
